src/test/files/RefOnlyInComplexTypesCheck.java

```diff
@@ -0,0 +1,17 @@
+/**
+ * This file is the sample code against we run our unit test.
+ * It is placed src/test/files in order to not be part of the maven compilation.
+ **/
+
+import org.mule.api.annotations.Connector;
+import org.mule.api.annotations.Processor;
+import org.mule.api.annotations.param.RefOnly;
+
+@Connector
+class RefOnlyInComplexTypesCheck {
+
+    @Processor // Noncompliant {{Processor 'failingMethod' contains complex types without @RefOnly.}}
+    public void failingMethod(SomeComplexType s1) {
+    }
+
```

Comment:
@urbanusjam Let's add some examples of Date, String, and BigDecimal. Those are in principle complex types but we let them pass for simple ones.